reference/system-architecture.md

```diff
@@ -0,0 +1,96 @@
+ See the [Docker Image](https://hub.docker.com/r/bitnami/etcd) and [Source on GitHub](https://GitHub.com/etcd-io/etcd).
+
+### RethinkDB
+ Database for permanent storage of PlaceOS configuration. 
+ Configuration: Three node cluster in 3 different availability zones within same region for [HA](#HADC).  
+ See the [Docker Image](https://hub.docker.com/_/rethinkdb) and [Source on GitHub](https://GitHub.com/rethinkdb/rethinkdb).
+
+### PlaceOS core
+ Application in a Docker container that interfaces with external devices/services.  
+ See the [Docker Image](https://hub.docker.com/r/placeos/core) and [Source on GitHub](https://GitHub.com/PlaceOS/core).
+
+### PlaceOS auth
+ Application in a Docker container that provides authentication.  
+ See the [Docker Image](https://hub.docker.com/r/placeos/auth) and [Source on GitHub](https://GitHub.com/PlaceOS/auth).
+
+### PlaceOS REST-API
+ Application in a Docker container that provides REST API for web applications.  
+ See the [Docker Image](https://hub.docker.com/r/placeos/rest-api) and [Source on GitHub](https://GitHub.com/PlaceOS/rest-api).
+
+### PlaceOS triggers
+ Perform user defined actions, without code, under certain conditions.  
+ See the [Docker Image](https://hub.docker.com/r/placeos/triggers) and [Source on GitHub](https://GitHub.com/PlaceOS/triggers).
+
+### PlaceOS dispatch
+ Reverse proxy incoming communications from devices/services to PlaceOS core. 
+ This is required to handle protocols like SNMP Traps - often not required.  
+ See the [Docker Image](https://hub.docker.com/r/placeos/dispatch) and [Source on GitHub](https://GitHub.com/PlaceOS/dispatch).
+
+### PlaceOS rubber-soul
+ Stream RethinkDB changes to Elasticsearch, which will index the documents for fast text searching.  
+ See the [Docker Image](https://hub.docker.com/r/placeos/rubber-soul) and [Source on GitHub](https://GitHub.com/PlaceOS/rubber-soul).
+
+## High Availability and Distributed Control {#HADC}
+High availability works by distributing services through Availability Zones. 
+For platforms like AWS Fargate, containers are initially automatically distributed.
+If a zone becomes unavailable, containers are re-launched in an alternative Availability Zone.
+
+<!-- possibly just link, here, maybe even push this stuff over to #key-concepts -->
+
+### Components
+**[Drivers](overview/drivers.md)** exist as files inside the PlaceOS core containers.
+PlaceOS Core creates a separate process for each Driver in use. 
+This will serve all the Modules (instances) of this Driver.
+
+**[Modules](overview/modules.md)** are instances of a Driver, that represent a single real-world device/service.
+They are distributed among the instances of core and exist in memory of the PlaceOS core containers.
+Modules reflect their runtime state in Redis, making it available for other components of the system
+
+The PlaceOS Backoffice web application manages them both via PlaceOS REST-API. 
+It updates module config via RethinkDB.
```

Comment:
Explanation of these may be better placed in their standalone articles, with a reference in the `PlaceOS Core` section above to the fact it is what runs these.

Reply:
Had a play around with it in different places - as I had a comment with similar thoughts to this too. But it feels a bit clunky in the `Core` section above, where those are otherwise all basically one-liners on the services. It *could* be added to the Overview pages for each, but I think it's good here because that's as minimal of an overview as is relevant for this page/concept. I'll think on it and lock something in tomorrow, open to feedback. Also @jeremyw24

Reply:
It reads correct where it is IMO